SignalServiceKit/src/Messages/OWSMessageSender.h

```diff
@@ -33,6 +33,23 @@ typedef void (^RetryableFailureHandler)(NSError *_Nonnull error);
 
 #pragma mark -
 
+NS_SWIFT_NAME(OutgoingAttachmentInfo)
+@interface OWSOutgoingAttachmentInfo : NSObject
```

Comment:
Unlike `SignalAttachment` (which I'd love to rename to `OutgoingAttachment`), this new class:

* Is a super lightweight struct with no logic.
* Resides in SSK (`SignalAttachment` resides in SM).
* Is MIME-type/`contentType`-oriented, not UTI-oriented.

This class just simplified the APIs as we pass around outgoing attachment metadata.

Reply:
This seems very similar to `OWSAttachmentInfo`, which so far was only used for `TSQuotedMessage`.

The difference I guess is that OWSAttachmentInfo is intended to be serialized into a YapDB object, whereas OWSOutgoingAttachmentInfo is an ephemeral object, maintaining a reference to a DataSource.

Even though they're very similar, I **think** it makes sense to have both.

Reply:
> Even though they're very similar, I think it makes sense to have both.

Good point.  Yeah, I think it makes sense to have both.